### ISSUE TITLE
compatibility issue with fluffy's worktab

### ISSUE THREAD

**Issue by missflora** [Aviuz/PrisonLabor#27]
#Exception in Verse.AI.ThinkNode_Priority TryIssueJobPackage: System.NullReferenceException: Object reference not set to an instance of an object
at WorkTab.VanillaWorkSettings.GetVanillaPriority (Verse.Pawn,Verse.WorkTypeDef) <0x00092>
at WorkTab.WorkPriorityTracker..ctor (Verse.Pawn,RimWorld.WorkGiverDef) <0x0003e>
at WorkTab.PawnPriorityTracker..ctor (Verse.Pawn) <0x00096>
at WorkTab.PriorityManager.get_Item (Verse.Pawn) <0x0004e>
at WorkTab.Pawn_Extensions.GetPriority (Verse.Pawn,RimWorld.WorkGiverDef,int) <0x00030>
at WorkTab.Pawn_WorkSettings_CacheWorkGiversInOrder/<>c__DisplayClass5_0.<Prefix>b__1 (RimWorld.WorkGiver) <0x0001d>
at System.Linq.Enumerable/<CreateWhereIterator>c__Iterator1D`1<RimWorld.WorkGiver>.MoveNext () <0x00100>
at System.Linq.Enumerable.Any<RimWorld.WorkGiver> (System.Collections.Generic.IEnumerable`1<RimWorld.WorkGiver>) <0x000a8>
at WorkTab.Pawn_WorkSettings_CacheWorkGiversInOrder.Prefix (RimWorld.Pawn_WorkSettings) <0x00173>
at (wrapper dynamic-method) RimWorld.Pawn_WorkSettings.CacheWorkGiversInOrder_Patch1 (object) <0x00014>
at RimWorld.Pawn_WorkSettings.get_WorkGiversInOrderNormal () <0x0001b>
at PrisonLabor.JobGiver_Labor.TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams) <0x004c9>
at Verse.AI.ThinkNode_Priority.TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams) <0x000a2>

Verse.Log:Error(String)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn, JobIssueParams)
RimWorld.ThinkNode_Conditional:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_SubtreesByTag:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.Pawn_JobTracker:DetermineNextJob(ThinkTreeDef&)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob_Patch1(Object, JobCondition, Boolean)
Verse.AI.Pawn_JobTracker:JobTrackerTick()
Verse.Pawn:Tick()
Verse.TickList:Tick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()

**Comment by kaeh** [comment]
Here's the github of the mod : https://github.com/FluffierThanThou/WorkTab

**Comment by Aviuz** [comment]
@missflora can you add info when this error occurs?

**Comment by kaeh** [comment]
Hey hmmm, i think i found a way to make a compatibility with WorkTab but it's kinda dirty.
@Aviuz , if you add WorkTab dll to project assemblies in local copy mode, you can update get_Pawns patch's Pawns method (line 208 iirc) from 
```C#
if (mainTabWindow is MainTabWindow_Work)
```
To
```C#
if (mainTabWindow is MainTabWindow_Work || mainTabWindow is WorkTab.MainTabWindow_WorkTab))
```

This way the WorkTab from Fluffy's mod is still present AND we have the prisoners in the list.

**Comment by Aviuz** [comment]
Yes this is solution but:
1. To attach .dll file I need approval of author.
2. This will make mod a lot bigger
3. It will probably be buggy if someone hasn't installed worktab

**Comment by Aviuz** [comment]
Oh you've got me idea

**Comment by kaeh** [comment]
Yup, I did this locally but it's temporary fix ^^. 
Oh? Can't wait to see that. 

_Sent from my HUAWEI WAS-LX1A using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Aviuz** [comment]
if (mainTabWindow is MainTabWindow_Work || mainTabWindow.GetType().ToString().Contains("MainTabWindow_WorkTab"))

**Comment by Aviuz** [comment]
Yeah I think it's working

**Comment by Aviuz** [comment]
fixed in 0.6
